### PR TITLE
Added custom user model with customer user manager

### PR DESCRIPTION
**Pull request by bartczak-pa**
This overwrites default django behaviour and allows to create users based on email instead of username.

Can we check if this code is okay?

### CODE REVIEW

**Comment by jacoor** [comment]
@bartczak-pa please see tests results. Those failed.
https://github.com/pymasterspl/reddit/actions/runs/9045129435/job/24854541143 
If you look here, it failed on migrations. This will be checked only after all is green. Please fix migrations.

**Comment by bartczak-pa** [comment]
@jacoor Migrations are fixed and code is passing tests now.

**Review by jacoor — CHANGES_REQUESTED** [review]
The code looks good.
However, Ruff is configured on purpose to enforce typings. 
Please, do fix ruff errors and avoid #noqa unless really necessary. 
I pushed an example of type updates for you, please fix other missing types. 
If you have issues with typing, you can try to use AI to fix 
- copilot - ask it to add typings for you (this works)
- or copy the code to Chat GPT and ask to add types.
Of course, do not forget to use `ruff check --fix` for other issues. 
Again, only use #noqa if this is really necessary, like:
`users/models.py:29:19: TRY003 Avoid specifying long messages outside the exception class`, however, for this one google has very good answer you can find after couple seconds search:
https://docs.astral.sh/ruff/rules/raise-vanilla-args/
(I put "how to fix TRY003 Avoid specifying long messages outside the exception class`" as a question to google).

File: users/models.py [hunk 1]
```diff
@@ -1,3 +1,41 @@
...
+from django.contrib.auth.base_user import BaseUserManager
+from django.contrib.auth.models import AbstractUser
+from django.db import models
 
-# Create your models here.
+
+class UserManager(BaseUserManager):
+    use_in_migrations: bool = True
+
+    def _create_user(self: "UserManager", email: str, password: str, **extra_fields: int) -> "User":
```
Comment (jacoor) [inline users/models.py:11]: Here you have an example of added typings.

File: users/models.py [hunk 2]
```diff
@@ -1,3 +1,41 @@
...
+        if not email:
+            message: str = "Users must have an email address"
```
Comment (jacoor) [inline users/models.py:13]: And fixed ruff errors, this one is EM101

**Comment by jacoor** [comment]
@bartczak-pa you need to resolve conflicts. After that, tests will start to run and we will know if the code good. 
Let me know if you need help.

**Review by jacoor — APPROVED** [review]

**Comment by jacoor** [comment]
Great work. Now merge away!

**Comment by bartczak-pa** [comment]
Typings and conflicts are resolved.

**Pull request merged**